Mathlib/RingTheory/HahnSeries/Basic.lean

```diff
@@ -220,6 +225,66 @@ instance [Nonempty Γ] [Nontrivial R] : Nontrivial (HahnSeries Γ R) :=
 
 section Order
 
+/-- The orderTop of a Hahn series `x` is a minimal element of `WithTop Γ` where `x` has a nonzero
+coefficient if `x ≠ 0`, and is `⊤` when `x = 0`. -/
+def orderTop (x : HahnSeries Γ R) : WithTop Γ :=
+  if h : x = 0 then ⊤ else x.isWF_support.min (support_nonempty_iff.2 h)
+
+@[simp]
+theorem orderTop_zero : orderTop (0 : HahnSeries Γ R) = ⊤ :=
+  dif_pos rfl
+
+theorem orderTop_of_ne {x : HahnSeries Γ R} (hx : x ≠ 0) :
+    orderTop x = x.isWF_support.min (support_nonempty_iff.2 hx) :=
+  dif_neg hx
+
+@[simp]
+theorem ne_zero_iff_orderTop {x : HahnSeries Γ R} : x ≠ 0 ↔ orderTop x ≠ ⊤ := by
+  constructor
+  · exact fun hx => Eq.mpr (congrArg (fun h ↦ h ≠ ⊤) (orderTop_of_ne hx)) WithTop.coe_ne_top
+  · contrapose!
+    simp_all only [orderTop_zero, implies_true]
+
+theorem zero_iff_orderTop {x : HahnSeries Γ R} : x = 0 ↔ orderTop x = ⊤ := by
```

Comment:
Intuitively, I would expect this lemma to be most useful in the other direction... possibly even as `simp`-lemma. But it would conflict with the `@[simp]` on the lemma above.